lib/pbench/server/sync.py

```diff
@@ -118,7 +118,10 @@ def update(
             if not message:
                 message = "ok"
 
-        Metadata.setvalue(dataset, Metadata.OPERATION, sorted(operations))
+        try:
+            Metadata.setvalue(dataset, Metadata.OPERATION, sorted(operations))
+        except Exception as e:
+            self.logger.warning("{} error updating ops: {}", dataset.name, str(e))
         if message:
             Metadata.setvalue(dataset, "server.status." + self.component, message)
```

Comment:
So...if the call at line 122 results in an exception, it seems real likely that the call at line 126 will result in one too...so shouldn't lines 125-126 either be inside the `try` block at line 121 or in a `try` block of their own?

Reply:
This was really part of the debugging, which I decided to leave in so we get a better clue if something goes wrong. Arguably a `raise` would be appropriate; the logging here is mostly to give a clearer idea of what/where went wrong.